### PR TITLE
fix(Component UI): Fixed SPDX License import functionality for XML CLI files

### LINKED ISSUE #636: Fix(component UI): spdx license import functionality to view and use spdx information is not working for XML CLI files.
###  for **`.xml`** (CLI) files: 

- Clicking on **`Show License Info`** button in _**Reelase -> Clearing Details**_ tab always gives error message: `Cannot load license information: parsererror (200).`

Attached screenshot for reference: 

image

### PR DESCRIPTION
**Pull request by akapti**
The below issue is fixed:
For **`XML CLI`** files, Clicking on **`Show License Info`** button in _**Reelase -> Clearing Details**_ tab always gives error message: `Cannot load license information: parsererror (200).`

Attached screenshot for reference:
**BEFORE:**
Before_01_LI

**STEP 1:**
After_01_LI

**STEP 2:**
After_02_LI

**AFTER:**
After_03_LI

closes #636 

Signed-off-by: Abdul Kapti <abdul.mannankapti@siemens.com>

### CODE REVIEW

**Comment by mcjaeger** [comment]
* for SPDX:
  * concluded licenses form the files
* for XXX:
  * only 'global' licenses
-> it is Ok, but there is a slight inconsistency then.

**Comment by nutanv1contr** [comment]
Hi @akapti ,
1. Tested the feature with one xml files and two rdf file. 
2. The license info listing for xml file is not working consistently. It is giving error as "Cannot load license information: parsererror (200)".
3. It is not able to read the rdf file all the time. It is giving message as "No license information found in the file." .
Could you please check again?

**Comment by mcjaeger** [comment]
I think there is maybe confusion with the file types, because the license info code checks for the file suffix.

**Comment by akapti** [comment]
@nutanv1contr I tested it again in my local, and `main license` is being displayed for `xml` CLI files.
I am not sure why rdf files are showing `No License found`. I haven't changed exsiting functionality w.r.t. `rdf` files.

**Comment by nutanv1contr** [comment]
Thanks for the insight @mcjaeger . Apparently that was a cache issue. When I tested in a fresh browser I am able to view the  license info results.   

ClearingDetailsLicenseinfo

**Comment by nutanv1contr** [comment]
Code looks good to me.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 1]
```diff
@@ -518,17 +519,29 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
...
         Set<String> concludedLicenseIds = new HashSet<>();
+        LicenseNameWithText licenseWithText = null;
         try {
             Release release = componentClient.getReleaseById(releaseId, user);
             List<LicenseInfoParsingResult> licenseInfoResult = licenseInfoClient.getLicenseInfoForAttachment(release,
                     attachmentContentId, user);
-            concludedLicenseIds = licenseInfoResult.stream()
-                    .flatMap(singleResult -> singleResult.getLicenseInfo().getConcludedLicenseIds().stream())
-                    .collect(Collectors.toSet());
+            if (attachmentName.endsWith(".rdf")) {
```
[the remark this answers is not in the document]
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: > file name ending 

the backend parser is being applied based on the same logic.
Kindly have a look at `isApplicableTo` method of `LicenseInfoParser` class, and let me know if there can be better approach for the same.

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 2]
```diff
@@ -518,17 +519,29 @@ private void loadSpdxLicenseInfo(ResourceRequest request, ResourceResponse respo
         User user = UserCacheHolder.getUserFromRequest(request);
         String releaseId = request.getParameter(PortalConstants.RELEASE_ID);
         String attachmentContentId = request.getParameter(PortalConstants.ATTACHMENT_ID);
+        String attachmentName = request.getParameter(PortalConstants.ATTACHMENT_NAME);
```
[the remark this answers is not in the document]
Comment (akapti) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: agreed with your point.
I felt passing the attachment name from UI would make it more effective then hitting the db just to get the attachment name.
anyways, I will make the required changes, and get the document from db.

**Pull request merged**